From: Kefeng Wang <wangkefeng.wang@huawei.com>
Subject: fs/proc/page: remove unneeded PageTail && PageSlab check
Date: Fri, 10 Nov 2023 11:33:18 +0800

Patch series "mm: remove page idle and young wrapper", v2.

Convert to use folio idle and young functions instead of page ones,
then remove all page idle and young wrapper.


This patch (of 7):

After commit dcb351cd095a ("page-flags: define behavior SL*B-related flags
on compound pages"), the slab could not be a tail, remove unneeded
PageTail && PageSlab check.

Link: https://lkml.kernel.org/r/20231110033324.2455523-1-wangkefeng.wang@huawei.com
Link: https://lkml.kernel.org/r/20231110033324.2455523-2-wangkefeng.wang@huawei.com
Signed-off-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Gregory Price <gregory.price@memverge.com>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/proc/page.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/fs/proc/page.c~fs-proc-page-remove-unneeded-pagetail-pageslab-check
+++ a/fs/proc/page.c
@@ -184,9 +184,6 @@ u64 stable_page_flags(struct page *page)
 	u |= kpf_copy_bit(k, KPF_LOCKED,	PG_locked);
 
 	u |= kpf_copy_bit(k, KPF_SLAB,		PG_slab);
-	if (PageTail(page) && PageSlab(page))
-		u |= 1 << KPF_SLAB;
-
 	u |= kpf_copy_bit(k, KPF_ERROR,		PG_error);
 	u |= kpf_copy_bit(k, KPF_DIRTY,		PG_dirty);
 	u |= kpf_copy_bit(k, KPF_UPTODATE,	PG_uptodate);
_
